test/modules/ModuleLib/TestOperationResultType.cs

```diff
@@ -9,6 +9,7 @@ public enum TestOperationResultType
         DirectMethod,
         Twin,
         Network,
-        Deployment
+        Deployment,
+        EdgeHubRestartMessageResult
```

Comment:
As I mentioned before, there is no relation that what properties we reported in DirectMethodTestResult will always need in EdgeHubRestart directmethod test result.  Therefore I think you should create a new type here and EdgeHubRestartDirectMethodResult should extend from TestResultBase.  If you think there is a relation before these 2 results, then at least create a new type for direct method edgehub restart scenario, and add 1 more constructor with TestOperationResultType argument in DirectMethodTestResult.